src/apps/allbridge/api.ts

```diff
@@ -0,0 +1,102 @@
+import got from 'got'
+import { NetworkId } from '../../types/networkId'
+import { Address } from 'viem'
+
+type SupportedAllbridgeChainSymbols =
+  | 'ETH'
+  | 'CEL'
+  | 'POL'
+  | 'ARB'
+  | 'OPT'
+  | 'BAS'
+
+const ALLBRIDGE_BLOCKCHAIN_SYMBOL_TO_NETWORK_ID: Record<
+  SupportedAllbridgeChainSymbols,
+  NetworkId
+> = {
+  ETH: NetworkId['ethereum-mainnet'],
+  CEL: NetworkId['celo-mainnet'],
+  POL: NetworkId['polygon-pos-mainnet'],
+  ARB: NetworkId['arbitrum-one'],
+  OPT: NetworkId['op-mainnet'],
+  BAS: NetworkId['base-mainnet'],
+}
+
+type AllbridgeApiResponse = {
+  [key in SupportedAllbridgeChainSymbols]: NetworkInfo
+}
+
+interface TokenInfo {
+  name: string
+  poolAddress: Address
+  tokenAddress: Address
+  decimals: number
+  symbol: string
+  poolInfo: PoolInfo
+  feeShare: string
+  apr: string
+  apr7d: string
+  apr30d: string
+  lpRate: string
+}
+
+interface PoolInfo {
+  aValue: string
+  dValue: string
+  tokenBalance: string
+  vUsdBalance: string
+  totalLpAmount: string
+  accRewardPerShareP: string
+  p: number
+}
+
+interface TransferTime {
+  allbridge: number
+  wormhole: number
+  cctp: number | null
+}
+
+interface TxCostAmount {
+  swap: string
+  transfer: string
+  maxAmount: string
+}
+
+interface NetworkInfo {
+  tokens: TokenInfo[]
+  chainId: number
+  bridgeAddress: Address
+  swapAddress: Address
+  transferTime: Record<SupportedAllbridgeChainSymbols, TransferTime>
+  confirmations: number
+  txCostAmount: TxCostAmount
+}
+
+export async function getAllbridgeTokenInfo({
+  networkId,
+}: {
+  networkId: NetworkId
+}): Promise<NetworkInfo> {
+  const tokenObj: Record<string, NetworkInfo> = {}
```

Comment:
if this function is for querying just a single network's token info, does it need to build a map from all tokens to networkInfo and then filter? Instead can't we just have a map from NetworkId to Allbridge chain symbol and just do `return allbridgeToken[NETWORK_ID_TO_ALLBRIDGE_CHAIN[networkId]]`?

Reply:
updated in [e20fa0c](https://github.com/valora-inc/hooks/pull/465/commits/e20fa0ceaa001902af80defbc199f28649013a63)